rxandroidble/src/main/java/com/polidea/rxandroidble/internal/operations/RxBleRadioOperationScan.java

```diff
@@ -46,8 +46,11 @@ public void run() {
     }
 
     public void stop() {
+
         // TODO: [PU] 29.01.2016 https://code.google.com/p/android/issues/detail?id=160503
-        rxBleAdapterWrapper.stopLeScan(leScanCallback);
+        if (startLeScanStatus) {
```

Comment:
I think you're right but such window is so small that it's almost impossible to run into such edge case. I think that `volatile` or a monitor will be enough. What about you @uKL ?